staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -671,13 +736,40 @@ func (c *Cacher) processEvent(event *watchCacheEvent) {
 }
 
 func (c *Cacher) dispatchEvents() {
+	// Jitter to help level out any aggregate load.
+	bookmarkTimer := c.clock.NewTimer(wait.Jitter(time.Second, 0.25))
+	// Stop the timer when watchBookmarkFeatureGate is not enabled.
+	if !c.watchBookmarkEnabled && !bookmarkTimer.Stop() {
+		<-bookmarkTimer.C()
+	}
+	defer bookmarkTimer.Stop()
+
+	lastResourceVersion := uint64(0)
+
 	for {
 		select {
 		case event, ok := <-c.incoming:
 			if !ok {
 				return
 			}
 			c.dispatchEvent(&event)
+			lastResourceVersion = event.ResourceVersion
+		case <-bookmarkTimer.C():
+			bookmarkTimer.Reset(wait.Jitter(time.Second, 0.25))
+			// Never send a bookmark event if we did not see an event here.
```

Comment:
Please add a comment that this is fine, because we don't provide any guarantees on sending bookmarks.